scene/3d/particles.cpp

```diff
@@ -47,6 +47,8 @@ PoolVector<Face3> Particles::get_faces(uint32_t p_usage_flags) const {
 void Particles::set_emitting(bool p_emitting) {
 
 	VS::get_singleton()->particles_set_emitting(particles, p_emitting);
+	if (p_emitting)
```

Comment:
Maybe check `one_shot` here too, to avoid setting internal process on non-one shot particles to never used it, nor disable it?

Reply:
Fair enough. I will also have to add a check in ``set_one_shot()`` that checks if ``emitting`` is true for the situation where ``emitting`` is already true when ``one_shot`` is set.

An alternative would be to change the check in ``_notification()`` to ``!one_shot || (one_shot && !is_emitting())``